src/services/ray/src/ray/deployments/controller/cluster/cluster.py

```diff
@@ -114,12 +113,15 @@ def update_nodes(self):
                 f"=> Node {name} updated with resources: {self.nodes[id].resources}"
             )
 
-        for node in self.nodes.keys():
-            if node not in current_nodes:
+        for node_id in self.nodes.keys():
+            if node_id not in current_nodes:
 
-                del self.nodes[node]
+                node = self.nodes.pop(node_id)
+                node.purge()
```

Comment:
Similar to what I commented regarding `deployment.delete()`, I think any exceptions that happen in purge should propagate up here, and at the very least be logged as an error before proceeding